src/common/AndroidUtils.ts

```diff
@@ -498,27 +523,47 @@ export class AndroidUtils {
                     );
                 }
 
-                if (AndroidUtils.isEmulatorAlreadyRunning(resolvedEmulator)) {
-                    // get port number from emu-launch-params.txt
-                    const portNumber = AndroidUtils.resolveEmulatorPort(
-                        resolvedEmulator,
-                        requestedPortNumber
-                    );
-                    return Promise.resolve(portNumber);
-                }
-
                 try {
+                    if (
+                        AndroidUtils.isEmulatorAlreadyRunning(resolvedEmulator)
+                    ) {
+                        // get port number from emu-launch-params.txt
+                        const portNumber = AndroidUtils.resolveEmulatorPort(
+                            resolvedEmulator,
+                            requestedPortNumber
+                        );
+
+                        const isWritable = await AndroidUtils.isEmulatorSystemWritable(
+                            portNumber
+                        );
+
+                        if (isWritable === writable) {
+                            // already launched and modes match so just return its port
+                            return Promise.resolve(portNumber);
+                        }
```

Comment:
The logic here ensure that if an emulator is already running then it must be running in the same "mode" as what is passed to this function. So if it is running with `writable-system` but this function is called with `writable=false` then that's a mismatch and we would shut down & relaunch.

Should we leave it this way or should we only care for the scenario where `writeable=true` but `isWritable=false`? In other words if an emulator is running with `writable-system` but this function is called with `writable=false` should we ignore and continue or should I do what I do now and shut it down + relaunch?

Reply:
Can you remind me why we are doing this? I remember last week you found that any modification done to `writable-system` gets wiped when the emulator is restarted without that flag. So wouldn't it be to our benefit to just let it keep running as a `writable-system`? Or you're trying to respect what the command says, meaning, if the configuration is different from previous run then shut down & relaunch?

Reply:
Yes that's exactly the case. I'm currently trying to respect the parameters of the command simply b/c what we have in core should be generic to be used by all repos and not tailored for a specific use case.

Reply:
> Should we leave it this way or should we only care for the scenario where writeable=true but isWritable=false?

I think we should only care about the case where we need a writeable file system but it's not booted that way. We can come back and change it later if we need to, but our requirement right now is simply requiring a writeable file system when we require it.